### PR TITLE
repokitteh: implement /retest on AZP

### LINKED ISSUE #820: Migrate Nighthawk's CI to machines with more memory
The current allocation only gives us 16GB of memory. `test_gcc` takes a very long time to finish and the integration tests can no longer execute.

We should migrate to machines with more memory.

### PR DESCRIPTION
**Pull request by mum4k**
Fixes #820.

### CODE REVIEW

**Comment by mum4k** [comment]
@jiajunye please review and assign back to me once done.

File: ci/repokitteh/modules/azure_pipelines.star [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        subchecks = [c for c in checks if c["external_id"] == check_id]
+        if len(subchecks) == 0:
+            continue
+
+        name_with_link = "[{}]({})".format(subchecks[0]["name"].split(" ")[0], _get_azp_link(check_id))
+
+        has_running = False
+        has_failure = False
+        for check in subchecks:
+            if check["conclusion"] != None and check["conclusion"] != "success":
```
Comment (jiajunye) [inline ci/repokitteh/modules/azure_pipelines.star:51]: nit: i would prefer is not None over != None

File: ci/repokitteh/modules/azure_pipelines.star [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+                has_failure = True
+            if check["status"] == "in_progress":
+                has_running = True
+
+        if not has_failure:
+            msgs += "Check {} didn't fail.\n".format(name_with_link)
+        else:
+            if has_running:
+                msgs += "Check {} isn't fully completed, but will still attempt retrying.\n".format(name_with_link)
```
Comment (jiajunye) [inline ci/repokitteh/modules/azure_pipelines.star:60]: just curious, isn't it more efficient to skip retrying a running check?

**Review by jiajunye — COMMENTED** [review]
LGTM

File: ci/repokitteh/modules/azure_pipelines.star (hunk 2 above)
Comment (mum4k) [reply ci/repokitteh/modules/azure_pipelines.star:60]: As noted above - this is copied from Envoy, choosing to keep it the same for an easier maintenance.

**Comment by mum4k** [comment]
(fyi) Going to turn this into a draft until we can test it. We might not even need to clone the file. Assuming we don't need to make any changes to it, we can just load it from Envoy's repo.

File: ci/repokitteh/modules/azure_pipelines.star (hunk 2 above)
Comment (jiajunye) [reply ci/repokitteh/modules/azure_pipelines.star:60]: Oops, agreed. I forgot the copy comment when I was reviewing the code.

File: repokitteh.star [hunk 3]
```diff
@@ -1,3 +1,6 @@
 use("github.com/repokitteh/modules/assign.star")
 use("github.com/repokitteh/modules/review.star")
 use("github.com/repokitteh/modules/wait.star")
+use("github.com/envoyproxy/nighthawk/ci/repokitteh/modules/azure_pipelines.star", secret_token = get_secret("azp_token"))
```
[the remark this answers is not in the document]
Comment (mum4k) [reply repokitteh.star]: Thank you for the advice, done.

**Comment by mum4k** [comment]
@jiajunye this now works and is ready for review. Please take another look.

**Comment by itayd** [comment]
lgtm fwiw

**Review by itayd — APPROVED** [review]

**Comment by jiajunye** [comment]
LGTM

**Pull request merged**